Expose ngx-datatable events; detailToggle, page, and sort

dlabrecq opened the pull request:
Expose ngx-datatable events; detailToggle, page, and sort. This will allow developers to more easily bypass the default implementation.

Note that these particular events are not compatible with the default pagination and sorting provided by pfng-table. Thus, I've updated the typescript docs to reflect that.

Also locked the typescript dev dependency at v2.6.2 in order for the Travis build to succeed. Using v2.7 breaks lodash.

dlabrecq committed:
fix(table): expose ngx-datatable events; detailToggle, page, and sort

elvisisking reviewed (approved):
Looks good @dlabrecq !

dlabrecq committed:
chore(typescript): v2.7 breaks lodash.

Locked the typescript dev dependency locked at v2.6.2 in order for the Travis build to succeed.
